test/interpreter/function-stdeva.spec.ts

```diff
@@ -0,0 +1,24 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {adr, detailedError} from '../testUtils'
+import {ErrorMessage} from '../../src/error-message'
+
+describe('Function STDEVA', () => {
+  it('should take at least two arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=STDEVA()'],
+      ['=STDEVA(1)']
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A2'))).toEqual(detailedError(ErrorType.DIV_BY_ZERO))
+  })
+
+  it('should calculate standard deviation (sample)', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=STDEVA(2, 3)'],
+      ['=STDEVA(B2:D2, E2, F2)', 2, 3, 'foo', 'bar', 4],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toBeCloseTo(0.707106781186548, 6)
+    expect(engine.getCellValue(adr('A2'))).toBeCloseTo(1.78885438199983, 6)
+  })
+})
```

Comment:
https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1018768_715980110

> **Unlike the STDEV function includes values of type Text and Logical. Text values are treated as number 0. Logical TRUE is treated as 1, and FALSE is treated as 0. Empty cells are not included.**
>
> The handling of string constants as parameters is implementation-defined. Either, string constants are converted to numbers, if possible and otherwise, they are treated as 0, or string constants are always treated as 0.

🙈  Which implementation did we choose? How does it relate to Excel requirements? 


- Arguments can be the following: numbers; names, arrays, or references that contain numbers; text representations of numbers; or logical values, such as TRUE and FALSE, in a reference.
- Arguments that contain TRUE evaluate as 1; arguments that contain text or FALSE evaluate as 0 (zero).
- If an argument is an array or reference, only values in that array or reference are used. Empty cells and text values in the array or reference are ignored.
- Arguments that are error values or text that cannot be translated into numbers cause errors.
- If you do not want to include logical values and text representations of numbers in a reference as part of the calculation, use the STDEV function.

Source: https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1018768_715980110

BTW We should add more tests for the requirements above

Reply:
> Which implementation did we choose? How does it relate to Excel requirements?

I don't really understand -- we implemented STDEVA function. The logic behind -A variants are nothing new, we already handled it with AVERAGEA, MAXA, MINA, SUMA functions.

Reply:
ODFF description suggests that STDEV can have different behavior depending on the implementation: "The handling of string constants as parameters is implementation-defined. Either, string constants are converted to numbers, if possible and otherwise, they are treated as 0, or string constants are always treated as 0."

Reply:
I see. Then for consistency I would keep it the same as in AVERAGEA. Where should it be documented?

Reply:
I would create a test that is our behaviour documentation with a clear description and an inconsistency comment